elephant_vending_machine/views.py

```diff
@@ -141,22 +143,75 @@ def upload_image():
             response = "Error with request: File extension not allowed."
     return response, response_code
 
-@APP.route('/log', methods=['GET'])
-def log():
-    """Returns the specified log file
+@APP.route('/experiment', methods=['POST'])
+def upload_experiment():
+    """Return string indicating result of experiemnt upload request
 
-    All requests sent to this route should have a log_name in
-    the query string, otherwise a 400 error will be returned
+    **Example request**:
 
-    Returns:
-        HTTP response 200 if log file exists, or HTTP response 500
-        if it does not.
+    .. sourcecode::
 
-    """
+      POST /experiment HTTP/1.1
+      Host: 127.0.0.1:5000
+      Content-Type: multipart/form-data; boundary=--------------------------827430006917349763475527
+      Accept-Encoding: gzip, deflate, br
+      Content-Length: 737067
+      Connection: keep-alive
+      ----------------------------827430006917349763475527
+      Content-Disposition: form-data; name="file"; filename="elephant.py"
+
+      <elephant.py>
+      ----------------------------827430006917349763475527--
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+      HTTP/1.0 200 OK
+      Content-Type: text/html; charset=utf-8
+      Content-Length: 21
+      Server: Werkzeug/0.16.1 Python/3.8.1
+      Date: Thu, 13 Feb 2020 15:35:32 GMT
+
+      Success: Image saved.
+
+    All requests sent to this route should have a python script file
+    included in the body of the request, otherwise a 400 error
+    will be returned
 
+    :status 201: file saved
+    :status 400: malformed request
+    """
     response = ""
-    if request.args.get('log_name') is not None:
-        response = 'This would be ' + request.args.get('log_name')
+    response_code = 400
+    if 'file' not in request.files:
+        response = "Error with request: No file field in body of request."
     else:
-        response = 'Error with request.'
-    return response
+        file = request.files['file']
+        if file.filename == '':
+            response = "Error with request: File field in body of response with no file present."
+        elif file and allowed_file(file.filename, ALLOWED_EXPERIMENT_EXTENSIONS):
+            filename = file.filename
+            save_path = os.path.dirname(os.path.abspath(__file__)) + EXPERIMENT_UPLOAD_FOLDER
+            file.save(os.path.join(save_path, filename))
+            response = "Success: Experiment saved."
+            response_code = 201
+        else:
+            response = "Error with request: File extension not allowed."
+    return response, response_code
+
+@APP.route('/log', methods=['GET'])
```

Comment:
Maybe we should return these as JSON with links to the files. We should probably just go ahead and make all of our responses that have a body return JSON actually since they'll eventually be consumed by a front-end. Open to discussion on this.

Reply:
If we plan to go with JSON, I'll have to change the way I coded my get-log endpoint. But for the long run it makes sense to do it with JSON so the next project can implement a front end.

Reply:
Should be a fast change. Flask [has some nice built in JSON features](https://www.programcreek.com/python/example/58915/flask.jsonify).

Reply:
Since it sounds like we may use our static folder to serve things anyway I'll go ahead and leave that there. Will switch to returning JSON

Reply:
@alexwaibel should I have it return links to Riya's endpoint? I.e. For file x.csv return base route + '/log/x.csv'?

Reply:
I think so? I've been googling a bit for this but haven't found a definitive answer yet. I envision this returning something like:

```
{
    "logs": ["<serverIP>/logs/x.csv", "<serverIP>/logs/y.csv", "<serverIP>/logs/z.csv"]
}
```

That way, the front end can simply list these as links and then when the user follows that link, the webserver should serve that file. Again, I'm not sure if there's a better way to handle this so if you have any alternate ideas we can discuss them.

Reply:
That's what I had in mind, and there's an easy way to grab the base url of the request, so this should be simple enough. I'll add the changes.

Reply:
You may as well convert all the other endpoints to return JSON real quick if we're going with that. Just return something like:
`
{
    "message":"<Current return message>"
}
`

Reply:
Done